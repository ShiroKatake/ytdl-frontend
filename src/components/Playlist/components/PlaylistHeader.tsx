interface IPlaylistHeaderProps {
  isCheckedAll: boolean;
  checkAll: (val: boolean) => void;
}

export const PlaylistHeader = ({isCheckedAll, checkAll}: IPlaylistHeaderProps) => {
  return (
    <tr>
      <th scope="col">#</th>
      <th scope="col">Thumbnail</th>
      <th scope="col">Title</th>
      <th scope="col">Download</th>
      <th scope="col">
        <>
          <label className="checkbox-label">Select All </label>
          <input
            type="checkbox"
            checked={isCheckedAll}
            id="selectAll"
            onChange={() => checkAll(!isCheckedAll)}
          />
        </>
      </th>
    </tr>
  );
};
