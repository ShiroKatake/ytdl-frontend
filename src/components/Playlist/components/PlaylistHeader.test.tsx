import { cleanup, render } from '@testing-library/react';
import { PlaylistHeader } from './PlaylistHeader';

describe('Testiemonial Page', () => {
  it('should match the snapshot', async () => {
    const component = render(
      <table>
        <tbody>
          <PlaylistHeader isCheckedAll={false} checkAll={() => {}}/>
        </tbody>
      </table>
    );
    
    expect(component.baseElement).toMatchSnapshot();
    cleanup();
  });
});
