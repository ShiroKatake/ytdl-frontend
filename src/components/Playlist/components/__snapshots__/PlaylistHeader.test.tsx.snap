// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testiemonial Page should match the snapshot 1`] = `
<body>
  <div>
    <table>
      <tbody>
        <tr>
          <th
            scope="col"
          >
            #
          </th>
          <th
            scope="col"
          >
            Thumbnail
          </th>
          <th
            scope="col"
          >
            Title
          </th>
          <th
            scope="col"
          >
            Download
          </th>
          <th
            scope="col"
          >
            <label
              class="checkbox-label"
            >
              Select All 
            </label>
            <input
              id="selectAll"
              type="checkbox"
            />
          </th>
        </tr>
      </tbody>
    </table>
  </div>
</body>
`;
